src/Qir/Tests/build-qir-tests.ps1

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+[CmdletBinding()]
+param (
+    [Parameter()]
+    [Switch]
+    $SkipQSharpBuild
+)
+
+& (Join-Path $PSScriptRoot .. .. .. build set-env.ps1)
+
+function Build-QirProject {
+    param (
+        [string]
+        $FolderPath
+    )
+
+    if (!$SkipQSharpBuild) {
+        Write-Host "##[info]Build Q# project for QIR tests '$FolderPath'"
+        dotnet build $FolderPath -c $Env:BUILD_CONFIGURATION -v $Env:BUILD_VERBOSITY
+        if ($LastExitCode -ne 0) {
+            Write-Host "##vso[task.logissue type=error;]Failed to compile Q# project at '$FolderPath' into QIR."
+            throw "Failed to compile Q# project at '$FolderPath' into QIR."
+        }
+    }
+}
+
+Write-Host "##[info]Compile Q# Projects into QIR"
+Build-QirProject (Join-Path $PSScriptRoot QIR-static qsharp)
+Build-QirProject (Join-Path $PSScriptRoot QIR-dynamic qsharp)
+Build-QirProject (Join-Path $PSScriptRoot QIR-tracer qsharp)
+Build-QirProject (Join-Path $PSScriptRoot FullstateSimulator qsharp)
+# Build-QirProject (Join-Path $PSScriptRoot samples StandaloneInputReference qsharp)
+
+
+Write-Host "##[info]Build QIR Tests"
+$oldCC = $env:CC
+$oldCXX = $env:CXX
+$oldRC = $env:RC
+
+$clangTidy = ""
+
+if (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
+{
+    Write-Host "On MacOS build QIR Tests using the default C/C++ compiler (should be AppleClang)"
+}
+elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
+{
+    Write-Host "On Linux build QIR Tests using Clang"
+    $env:CC = "clang-11"
+    $env:CXX = "clang++-11"
+    $env:RC = "clang++-11"
+    $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy-11"
+}
+elseif (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
+{
+    Write-Host "On Windows build QIR Tests using Clang"
+    $env:CC = "clang.exe"
+    $env:CXX = "clang++.exe"
+    $env:RC = "clang++.exe"
+
+    if (!(Get-Command clang -ErrorAction SilentlyContinue) -and (choco find --idonly -l llvm) -contains "llvm") {
+        # LLVM was installed by Chocolatey, so add the install location to the path.
+        $env:PATH += ";$($env:SystemDrive)\Program Files\LLVM\bin"
+    }
+
+    if (Get-Command clang-tidy -ErrorAction SilentlyContinue) {
+        # Only run clang-tidy if it's installed. This is because the package used by chocolatey on
+        # the build pipeline doesn't include clang-tidy, so we allow skipping that there and let
+        # the Linux build catch tidy issues.
+        $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy"
+    }
+} else {
+    Write-Host "##vso[task.logissue type=error;]Failed to identify the OS. Will use default CXX compiler"
+}
```

Comment:
(Minor)
I see this also in more than one place. Consider extracting into a common file invoked from other files.

Reply:
This one I agree should be done now, as we are likely to need to make changes to how cmake is invoked and don't want to have to do that in too many places. I'll work in this now.